app/helpers/application_helper.rb

```diff
@@ -76,6 +76,38 @@ def format_datetime(date, format = :long)
   def time_period_class(item)
     return 'not-yet-valid' if item.not_yet_valid?
     return 'currently-valid' if item.currently_valid?
-    return 'expired' if item.expired?
+    'expired' if item.expired?
+  end
+
+  # Generates a page header. The title shown will be the +.header+ key in the page
+  # that calls this helper.
+  #
+  # @yield A block in which other helper methods may be called, to place child elements
+  #   on the far right of the header.
+  # @return [String]
+  def page_header
+    content_tag(:div, class: 'page-header') do
+      content_tag(:h1) do
+        content_tag(:span, t('.header')) +
+          content_tag(:div, class: 'pull-right') do
+            yield if block_given?
+          end
+      end
+    end
+  end
+
+  # Generates all page titles, from the reverse breadcrumb if it's available,
+  # otherwise checks the +content_for?+ Rails helper. Appends the default
+  # title to everything.
+  # @return [String]
+  def page_title
+    if !breadcrumb_names.empty?
```

Comment:
I think this should be reversed. content_for should be checked first, then breadcrumbs, then coursemology branding.

I'm also not sure i18n-wise if there's a way to be nice about this. Not every script uses a dash.